cpp/daal/src/algorithms/normalization/zscore/zscore.cpp

```diff
@@ -95,7 +96,7 @@ Result::Result() : daal::algorithms::Result(lastResultId + 1)
 Result::Result(const Result & o)
 {
     ResultImpl * pImpl = dynamic_cast<ResultImpl *>(getStorage(o).get());
-    DAAL_ASSERT(pImpl);
+    assert(pImpl);
```

Comment:
DAAL_ASSERT only interrupts execution if project is compiled with debug options so it can not be used for important checks (like user input validation)
https://github.com/oneapi-src/oneDAL/blob/d4c42cfcce9f26c3e8c7d251c392f97ca526c05e/cpp/daal/include/services/daal_defines.h#L475